dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/EventManager.java

```diff
@@ -194,9 +203,14 @@ public ImportSummaries updateEvents( final List<Event> events, final WorkContext
         // pre-process events
         executorsByPhase.get( EventProcessorPhase.UPDATE_PRE ).execute( workContext, events );
 
-        importSummaries.addImportSummaries(
-            // Run validation against the remaining "updatable" events //
-            updateValidatingEventChecker.check( workContext, events ) );
+        if ( ImportStrategyUtils.isUpdate( workContext.getImportOptions().getImportStrategy() ) )
+        {
+            importSummaries.addImportSummaries( run( workContext, events, checkersRunOnUpdate ) );
+        }
+        else
+        {
+            importSummaries.addImportSummaries( Collections.emptyList() );
+        }
```

Comment:
These `else` blocks look odd. This should be a NOOP or the semantics are badly defined and should be changed. Adding an empty list should at last not do anything important.

Reply:
I am also not satisfied with how ImportSummaries are gathered, but I don't understand that part good enough yet to make a suggestion.

The intention of my PR is to inline, move code. I haven't at least as far as I know changed logic.

The EventManager
https://github.com/dhis2/dhis2-core/blob/f8054738d719c2d5251de87bf7d55108db2c13cd/dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/EventManager.java#L124-L126

calls the ValidatingEventChecker
https://github.com/dhis2/dhis2-core/blob/f8054738d719c2d5251de87bf7d55108db2c13cd/dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/ValidatingEventChecker.java#L56

which calls the https://github.com/dhis2/dhis2-core/blob/f8054738d719c2d5251de87bf7d55108db2c13cd/dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/EventImporterValidationRunner.java#L55
only in case the import strategy is supported

if not it adds an empty list to the import summaries.

If I understood the flow correctly 😅

Reply:
will remove the else blocks as suggested 👍🏻